challenge_2/cpp/sven/src/Single.cpp

```diff
@@ -0,0 +1,22 @@
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <algorithm>
+
+int main(int argc, char* argv[])
+{
+	std::unordered_map<std::string, bool> map;
+	for (int i =1; i < argc; ++i)
+	{
+		std::string el = argv[i];
+		if (map.find(el) == map.cend())
+		{
+			map[el] = true;
+		}
+		else
+		{
+			map[el] = false;
+		}
+	}
+	std::cout << std::find_if(map.cbegin(), map.cend(), [](const std::unordered_map<std::string, bool>::value_type& vt) { return vt.second; })->first << std::endl;
```

Comment:
Before this statement, we've build the map that contains the inputvalues as keys of the map and whether they appear once or multiple times as bool values of the map (once = true, multiple occurrences = false).

So, the only thing that is left to be done is to go trough the map and look for the first (and single) value that is "true" and return its key (number):
find_if is a method present in the <algorithm> class which goes trough a given range and returns the first element it finds for which the given predicate is true.

I use find_if by giving the entire range of the map (map.cbegin(), map.cend()) and simply use the value of the map as predicate (as said before: the value of the map will be true if the value only appeared once in the given input).
find_if returns the correct element and that's the output that we print out on the standard output.

let's go trough it step by step:
std::find_if takes 3 arguments
- the first is the beginning of the range. I use map.cbegin(), the beginning of our map
- the second is the end of the range. map.cend()
- the last argument is the predicate that is applied on each element. 
I use a lambda as predicate.
The lambda takes a single argument: "vt" of type "const std::unordered_map<std::string, bool>::value_type&" (the type of a value in an unordered_map), and returns the associated value of the element "{ return vt.second; }"

the "->first" that comes after the find_if is to retrieve the key of the found element and print it out